src/mscorlib/shared/System/Security/SecurityException.cs

```diff
@@ -45,7 +47,6 @@ public SecurityException(string message, Type type, string state)
         protected SecurityException(SerializationInfo info, StreamingContext context)
             : base(info, context)
         {
```

Comment:
This type is custom serialized on desktop. It needs to be here. 

Some of the properties we don't have (were snipped due to dependencies): Action, FirstPermissionThatFailed, Zone. Those should all serialize hard coded values: two of them are enums. You probably want to make dummy private enums right here, with just the default value you want, so as to serialize them?

Reply:
> You probably want to make dummy private enums right here,

I doubt that this is going to work. I think we should give up on serializing the details of SecurityException.

Note that the SecurityException deserialization code on desktop has a big try/catch around it, so even if the details are missing, the SecurityException is still going to deserialize fine with correct message, etc.

Reply:
I guess we are ok then as long as deserialization uses TryGetValue or a catch block. Or the type is a reference or primitive type where we can just use null or 0.

Reply:
I wasn't sure if reference types that aren't serializable would work because of their AssemblyQualifiedName. But it seems if you pass null into it the type AQN won't be put into the serialization blob which is great!

Reply:
Won't there be a problem deserializing these from desktop?

Reply:
You are right. It's the same as SqlException for which I disabled Desktop --> Core serialization in the tests. We can't support SecurityException without changing the netfx code as it serializes type AssemblyName which we don't want to support. I think it's most important that we document the limitations when it comes to Exception serialization (unsupported and partially supported types).